.circleci/chain_on_pr.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash -e
+
+#touch $BASH_ENV
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | bash
+
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+
+nvm install v10.14.1
+cd .circleci && npm install @orbs-network/orbs-nebula && cd ..
+
+# Installing aws cli
+echo "Installing AWS CLI"
+sudo apt-get update
+sudo apt-get install -y python-dev
+sudo apt-get install -y python-pip
+sudo pip install awscli
+
+aws --version
+
+COMMIT_HASH=$(./docker/hash.sh)
+
+# Determine if we have an active PR
+# (The condition means if the variable is not empty)
+if [ ! -z "$CI_PULL_REQUESTS" ]
+then
+    echo "We have an active PR ($CI_PULL_REQUESTS)"
+
+    echo "Downloading the current testnet Boyar config.json"
+    curl -O https://boyar-testnet-bootstrap.s3-us-west-2.amazonaws.com/boyar/config.json
+    echo "Done downloading!"
+
+    echo "Creating a network for this PR within the config.json file.."
+    PR_CHAIN_ID=$(node .circleci/testnet-deploy-new-chain-for-pr.js $CI_PULL_REQUESTS $COMMIT_HASH)
+    echo "Done adding a new chain ($PR_CHAIN_ID)"
+
+    echo "Copying the newly updated config.json to S3"
+    aws s3 cp --acl public-read config.json s3://boyar-testnet-bootstrap/boyar/config.json
+    echo "Done!"
+
+    echo "Configuration updated, waiting for the new PR chain ($PR_CHAIN_ID) to come up!"
+
+    echo "Sleeping for 10 minutes to allow the network to come up"
+    sleep 300
+    echo "Still sleeping..."
+    sleep 400
+
+    echo "Checking deployment status:"
+    node .circleci/check-testnet-deployment.js $PR_CHAIN_ID
+
+    echo "Running the E2E suite against the newly deployed isolated chain for this PR.."
+    export API_ENDPOINT=http://35.161.123.97/vchains/$PR_CHAIN_ID/ \
+        STRESS_TEST_NUMBER_OF_TRANSACTIONS=100 \
+        STRESS_TEST_FAILURE_RATE=20 \
+        STRESS_TEST_TARGET_TPS=200 \
+        STRESS_TEST=true
+
+    ./git-submodule-checkout.sh
```

Comment:
I would move it to a separate step in `config.yml` as usual. It is confusing to have it here.

Reply:
Are you talking about running the vendoring script or running the tests ? or both?
My reason for not having separate steps in the `config.yml` is due to the fact the terminal state gets discarded between `-run:` lines. So the fact it's all in 1 script makes it easier for me to maintain the state of having NVM installed and the correct Node.js version used , same goes for AWS CLI in case it sets any environment variables.

In order to move these parts of running the tests to a separate line in the configuration I would need to break this script apart to what happens before the tests are run and what happens after it. I don't find it that important.

Can you elaborate why you think it's confusing?

Reply:
I would like you to move this specific line: `./git-submodule-checkout.sh` (I am not sure we actually need to do this at all). Nothing seems to depend on these modules.

Reply:
Moved @netoneko